src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -1,50 +1,218 @@
+  void addTestData() {
+    int numberOfUsers = ThreadLocalRandom.current().nextInt(10, 101);
+    int numberOfConversations = ThreadLocalRandom.current().nextInt(5, 21);
+    int numberOfMessages = ThreadLocalRandom.current().nextInt(100, 1001);
+
+    User newUser;
+    for (int i = 0; i < numberOfUsers; i++) {
+      newUser = new User(UUID.randomUUID(), "user"+i, "password",
+        Instant.ofEpochSecond(ThreadLocalRandom.current().nextInt(1, 10001)));
+
+      userStore.addUser(newUser);
     }
 
-    @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response)
-        throws IOException, ServletException {
+    Conversation newConversation;
+    User owner;
+    for (int i = 0; i < numberOfConversations; i++) {
+      owner = userStore.getAllUsers().get(ThreadLocalRandom.current().nextInt(0, userStore.getAllUsers().size()));
+      newConversation = new Conversation(UUID.randomUUID(), owner.getId(), "conversation"+i,
+        Instant.now());
+
+      conversationStore.addConversation(newConversation);
+    }
 
-        String user = (String) request.getSession().getAttribute("user");
+    Message newMessage;
+    Conversation conversation;
+    User author;
+    int numberOfWords;
+    String content;
+    for (int i = 0; i < numberOfMessages; i++) {
+      author = userStore.getAllUsers().get(ThreadLocalRandom.current().nextInt(0, userStore.getAllUsers().size()));
+      conversation = conversationStore.getAllConversations().get(ThreadLocalRandom.current().
+        nextInt(0, conversationStore.getAllConversations().size()));
 
-        if (admins.contains(user)) {
-            request.getRequestDispatcher("/WEB-INF/view/admin.jsp").forward(request, response);
-        } else {
-            response.sendError(HttpServletResponse.SC_NOT_FOUND);
-        }
+      numberOfWords = ThreadLocalRandom.current().nextInt(0, 21);
+      content = "" + (char) ThreadLocalRandom.current().nextInt('a', 'z'+1);
+      for (int w = 1; w < numberOfWords; w++) {
+        content += " " + (char) ThreadLocalRandom.current().nextInt('a', 'z'+1);
+      }
 
+      newMessage = new Message(UUID.randomUUID(), conversation.getId(), author.getId(),
+       content, Instant.now());
+
+      messageStore.addMessage(newMessage);
     }
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+    throws IOException, ServletException {
 
-    @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response)
-        throws IOException, ServletException {
+    String user = (String) request.getSession().getAttribute("user");
 
+    if (admins.contains(user)) {
+      computeAdminStats(request);
+      request.getRequestDispatcher("/WEB-INF/view/admin.jsp").forward(request, response);
+    } else {
+      response.sendError(HttpServletResponse.SC_NOT_FOUND);
     }
+
+  }
+
+  /*
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
```

Comment:
Since this isn't being used, you should delete it, not just comment it out: https://blog.kentcdodds.com/please-don-t-commit-commented-out-code-53d0b5b26d5f